pyteal/ast/subroutine.py

```diff
@@ -183,29 +279,29 @@ def has_return(self):
 
 class SubroutineCall(Expr):
     def __init__(
-        self, subroutine: SubroutineDefinition, args: List[Union[Expr, ScratchVar]]
+        self,
+        subroutine: SubroutineDefinition,
+        args: List[Union[Expr, ScratchVar, abi.BaseType]],
     ) -> None:
         super().__init__()
         self.subroutine = subroutine
         self.args = args
 
         for i, arg in enumerate(args):
-            arg_type = None
-
-            if not isinstance(arg, (Expr, ScratchVar)):
+            if isinstance(arg, Expr):
+                arg_type = arg.type_of()
+            elif isinstance(arg, ScratchVar):
+                arg_type = arg.type
+            elif SubroutineDefinition.is_abi_class(type(arg)):
```

Comment:
I think `isinstance(arg, abi.BaseType)` would be the same as this check, and if you do this you could avoid the cast on the next line